### PR TITLE
Update run-number.md

### PR DESCRIPTION
**Pull request by StephenHodgson**
Added Expressions section to document their usage in the build number

### CODE REVIEW

**Comment by PRMerger10** [comment]
@StephenHodgson : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger10** [comment]
@StephenHodgson : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/pipelines/process/run-number.md [hunk 1]
```diff
@@ -99,6 +99,14 @@ $(Build.DefinitionName)_$(Build.DefinitionVersion)_$(Build.RequestedFor)_$(Build
 
 The first four variables are predefined. `My.Variable` is defined by you on the [variables tab](variables.md).
 
+### Expressions
+
+You can also use [Expressions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops) to dynamically change the build version number depending on certain conditions. For example, if the build is triggered from a pull request, then we can use that in the number format, otherwise fall back to the builds source branch name, or even date format if this variable is also not available:
+
+```yaml
+name: ${{ coalesce( variables['System.PullRequest.PullRequestNumber'], '$(Build.SourceBranchName)', '$(Date:yyyyMMdd)') }}$(Rev:.r)
```
Comment (StephenHodgson) [inline docs/pipelines/process/run-number.md:107]:
```suggestion
name: ${{ coalesce( '$(System.PullRequest.PullRequestNumber)', '$(System.PullRequest.SourceBranch)', '$(Date:yyyyMMdd)') }}$(Rev:.r)
```

**Comment by PRMerger17** [comment]
@StephenHodgson : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/pipelines/process/run-number.md [hunk 2]
```diff
@@ -99,6 +99,14 @@ $(Build.DefinitionName)_$(Build.DefinitionVersion)_$(Build.RequestedFor)_$(Build
 
 The first four variables are predefined. `My.Variable` is defined by you on the [variables tab](variables.md).
 
+### Expressions
+
+You can also use [Expressions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops) to dynamically change the build version number depending on certain conditions. For example, if the build is triggered from a pull request, then we can use that in the number format, otherwise fall back to the builds source branch name, or even date format if this variable is also not available:
+
+```yaml
+name: ${{ coalesce( '$(System.PullRequest.PullRequestNumber)', '$(System.PullRequest.SourceBranch)', '$(Date:yyyyMMdd)') }}$(Rev:.r)
```
Comment (StephenHodgson) [inline docs/pipelines/process/run-number.md:107]: Seems as though `$(System.PullRequest.PullRequestNumber)` doesn't collapse into a null string. This example may not work correctly.

https://github.com/microsoft/azure-pipelines-yaml/issues/474

Comment (StephenHodgson) [reply docs/pipelines/process/run-number.md:107]: I was eventually able to get this to work using:
```
variables:
  ${{ if eq( variables['build.reason'], 'PullRequest' ) }}:
    date: ''
  ${{ if ne( variables['build.reason'], 'PullRequest' ) }}:
    date: ${{ '$(Date:yyyyMMdd)' }}

name: ${{ coalesce( variables['date'], '$(System.PullRequest.PullRequestNumber)', '$(Date:yyyyMMdd)' ) }}$(Rev:.r)
```

**Pull request closed without merging**